### PR TITLE
Landing Page & Signing

### PR DESCRIPTION
**Pull request by MADahab**
-Change home to be the main Route, rending Nav bar and Outlet under it

-Create Landing Page

-Create Sign up Page

-Create Login Page

### CODE REVIEW

File: src/pages/Home.tsx [hunk 1]
```diff
@@ -1,7 +1,116 @@
...
+            <div>
+              <Link to="/login" style={styles.smallerButton}>Login</Link>
+              <Link to="/signup" style={styles.smallerButton}>Sign Up</Link>
+            </div>
+          )}
+        </div>
+      </nav>
+      <Outlet />
+    </div>
+  );
```
Comment (amrAaboelnaga) [inline src/pages/Home.tsx]: nav bar or header bar is usually a file on its own, usually in components.

File: src/pages/Home.tsx [hunk 2]
```diff
@@ -1,7 +1,116 @@
...
+  },
+  logOutButton: {
+    backgroundColor: '#333',
+    color: '#fff',
+    border: 'none',
+    padding: '6px 15px',
+    cursor: 'pointer',
+    borderRadius: '15px',
+    fontSize: '1rem',
+    marginLeft: '10px',
```
Comment (amrAaboelnaga) [inline src/pages/Home.tsx]: consistency is key, use either rem or px through out the app, it doesnt matter which, but lets be consistent.

File: src/pages/LandingPage.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+          <div>
+            <h3 onClick={() => navigate(aboutLink)}>About</h3>
+            <p>
+              <span style={styles.link} onClick={() => navigate(aboutUsLink)}>About Us</span><span> | </span>
+              <span style={styles.link} onClick={() => navigate(privacyPolicyLink)}>Privacy Policy</span><span> | </span>
+              <span style={styles.link} onClick={() => navigate(termsOfServiceLink)}>Terms of Service</span>
+            </p>
+          </div>
+        </div>
+      </footer>
```
Comment (amrAaboelnaga) [inline src/pages/LandingPage.tsx]: should be in a seperate file in components same as header.

File: src/pages/Login.tsx [hunk 4]
```diff
@@ -5,30 +5,115 @@ import { handleLogin } from '../handlers/authHandler';
...
+    </div>
   );
 });
 
+const styles = {
+  container: {
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    height: 'calc(100vh - 50px)',
```
Comment (amrAaboelnaga) [inline src/pages/Login.tsx]: should be calculated in the root dom, then the rest is flex

File: src/routes/AppRoutes.tsx [hunk 5]
```diff
@@ -4,16 +4,19 @@ import Home from '../pages/Home';
...
   return (
     <Router>
       <Routes>
-        <Route path="/" element={<Home />} />
-        <Route path="/login" element={<Login />} />
-        <Route path="/signup" element={<Signup />} />
-        <Route path="/dashboard" element={<Dashboard />} />
-        {/* Add more routes as needed */}
+        <Route path="/" element={<Home />} >
+          <Route path="/landingpage" element={<LandingPage />} />
```
Comment (amrAaboelnaga) [inline src/routes/AppRoutes.tsx:17]: landing page should be / which is the home technically.

File: src/pages/Home.tsx (hunk 1 above)
Comment (MADahab) [reply src/pages/Home.tsx]: Done

File: src/pages/Home.tsx (hunk 2 above)
Comment (MADahab) [reply src/pages/Home.tsx]: Done

File: src/pages/Login.tsx (hunk 4 above)
Comment (MADahab) [reply src/pages/Login.tsx]: Done

File: src/routes/AppRoutes.tsx (hunk 5 above)
Comment (MADahab) [reply src/routes/AppRoutes.tsx:17]: as agreed to let the home has nav and footer and redirect to Landing or Dashboard based on if user logged in or not

**Pull request merged**